src/main/java/eu/dissco/backend/controller/BaseController.java

```diff
@@ -26,6 +35,34 @@ public abstract class BaseController {
   protected final ObjectMapper mapper;
   private final ApplicationProperties applicationProperties;
 
+  protected User getUser(Authentication authentication) throws ForbiddenException {
+    var tokenEncoded = ((Jwt) authentication.getPrincipal()).getTokenValue();
+    var tokenBody = tokenEncoded.split("\\.")[1];
+    var tokenString = Base64.getUrlDecoder().decode(tokenBody);
+    JsonNode token;
+    try {
+      token = mapper.readTree(tokenString);
+    } catch (IOException e) {
+      log.error("Unable to read authentication token");
+      throw new ForbiddenException("Unable to read authentication token");
+    }
+    if (token.get("orcid") != null) {
+      StringBuilder fullName = new StringBuilder();
+      if (token.get("given_name") != null){
+        fullName.append(token.get("given_name").asText())
+            .append(" ");
```

Comment:
Not sure about the append here, if family name is not there it might be an issue. Maybe trim the name before adding it could solve it, or check if full_name is not empty when adding the family name